cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/batch/MockSource.java

```diff
@@ -86,13 +98,34 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     context.setInput(Input.ofDataset(config.tableName));
   }
 
+  /**
+   * Get the plugin config to be used in a pipeline config. If the source outputs records of the same schema,
+   * {@link #getPlugin(String, Schema)} should be used instead, so that the source will set an output schema.
+   *
+   * @param tableName the table backing the mock source
+   * @return the plugin config to be used in a pipeline config
+   */
   public static ETLPlugin getPlugin(String tableName) {
     Map<String, String> properties = new HashMap<>();
     properties.put("tableName", tableName);
     return new ETLPlugin("Mock", BatchSource.PLUGIN_TYPE, properties, null);
   }
 
   /**
+   * Get the plugin config to be used in a pipeline config. The source must only output records with the given schema.
+   *
+   * @param tableName the table backing the mock source
+   * @param schema the schema of records output by this source
+   * @return the plugin config to be used in a pipeline config
+   */
+  public static ETLPlugin getPlugin(String tableName, Schema schema) {
```

Comment:
i meant that method can call this method `getPlugin(tableName, null)` anyway its not critical.